src/posix/platform/spi_interface.cpp

```diff
@@ -695,14 +699,13 @@ otError SpiInterface::WaitForFrame(const struct timeval &aTimeout)
 
             // Read event data to clear interrupt.
             VerifyOrDie(read(mIntGpioValueFd, &event, sizeof(event)) != -1, OT_EXIT_FAILURE);
+            isDataReady = CheckInterrupt();
         }
+    }
 
-        // If we can receive a packet.
-        if (CheckInterrupt())
-        {
-            otLogDebgPlat("WaitForFrame(): Interrupt.");
-            PushPullSpi();
-        }
```

Comment:
How about update the code like this? Then we don't need to define new variable and the code logic is easier.
```
if (ret >= 0) 
{ 
    if ((ret == 0) && ((timeout.tv_sec != 0) || (timeout.tv_usec != 0)))
    {
        ExitNow(error = OT_ERROR_RESPONSE_TIMEOUT);
    }
  
    if (FD_ISSET(mIntGpioValueFd, &readFdSet))
    {     
        struct gpioevent_data event;
    
        // Read event data to clear interrupt.
        VerifyOrDie(read(mIntGpioValueFd, &event, sizeof(event)) != -1, OT_EXIT_FAILURE);
    }     
              
    // If we can receive a packet.  
    if (CheckInterrupt())
    {     
        otLogDebgPlat("WaitForFrame(): Interrupt.");
        PushPullSpi();
    }     
}      
else if (errno != EINTR)
{             
    DieNow(OT_EXIT_ERROR_ERRNO);
}
```